File: directorysetup.c
#include <stdio.h>
#include <stdlib.h>
#include <direct.h>
#include <dirent.h>

int main(void) {
    const char *path = "newdir";

    if (_mkdir(path) == 0) {
        printf("Directory '%s' created successfully.\n", path);
    } else {
        perror("mkdir");
    }

    printf("\nContents of current directory:\n");
    DIR *dir = opendir(".");
    if (dir == NULL) {
        perror("opendir");
        return EXIT_FAILURE;
    }

    struct dirent *entry;
    int found = 0;
    while ((entry = readdir(dir)) != NULL) {
        printf("  %s\n", entry->d_name);
        found = 1;
    }
    closedir(dir);

    if (!found) {
        printf("  (no files found)\n");
    }

    printf("\nDirectory listing complete.\n");
    return EXIT_SUCCESS;
}

File: DirectorySetup.java
import java.io.IOException;
import java.nio.file.DirectoryStream;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class DirectorySetup {

    public static void main(String[] args) {
        String path = args.length > 0 ? args[0] : ".";

        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(path))) {
            System.out.printf("Contents of directory '%s':%n", path);
            for (Path entry : stream) {
                System.out.println("  " + entry.getFileName());
            }
        } catch (IOException e) {
            System.err.println("opendir/readdir/closedir: " + e.getMessage());
        }

        System.out.println("Directory listing complete.");
    }
}
